### DISCUSSION TITLE
Element Consistency

### DISCUSSION THREAD

**Discussion by eliotwrobson** [PrairieLearn/PrairieLearn#7353] [Dev]
Discussed with @firasm on https://github.com/PrairieLearn/PrairieLearn/issues/7345#issuecomment-1471418752. The purpose of this discussion is to collect / hash out issues related to elements behaving and displaying similarly. Some examples of these issues are here:

- https://github.com/PrairieLearn/PrairieLearn/issues/7345
- https://github.com/PrairieLearn/PrairieLearn/issues/6968
- https://github.com/PrairieLearn/PrairieLearn/issues/878
- https://github.com/PrairieLearn/PrairieLearn/pull/5901
- https://github.com/PrairieLearn/PrairieLearn/issues/6350
- https://github.com/PrairieLearn/PrairieLearn/issues/5369
- https://github.com/PrairieLearn/PrairieLearn/issues/7385
- https://github.com/PrairieLearn/PrairieLearn/issues/2344
- https://github.com/PrairieLearn/PrairieLearn/issues/5902
- https://github.com/PrairieLearn/PrairieLearn/issues/1671
- https://github.com/PrairieLearn/PrairieLearn/issues/5540 

In the discussions below, we can decide which of these should or should not be implemented, and the semantics for each. Some of them have already been partially implemented (such as `allow-blank`), and we should remain consistent with the partial work that has been done.

**Comment by firasm** [comment]
Thanks @eliotwrobson - I wonder if it would be more efficient to jump on a zoom call and go through some of these and get to a nice starting point so others can more easily contribute?

**Reply by eliotwrobson** [reply]
I think that's a good idea, maybe in one of the upcoming office hours we can spend a few minutes on some of these? I don't think the discussions will be _too_ long, most of these items are relatively light.

**Reply by firasm** [reply]
Ok I'll plan to be there on Thursday this week.

**Reply by eliotwrobson** [reply]
@firasm just as a heads up, since this PR got delayed because of finals starting soon, it's a good candidate to start making the changes outlined in this discussion: https://github.com/PrairieLearn/PrairieLearn/pull/6980

**Comment by firasm** [comment]
Based on discussion in Mar 23 PL Office hours, we decided:

- [ ] Definitely show `invalid` submissions in the question-panel for all elements
- [ ] Converge on `sort` for specifying order (#5369)
- [ ] Correctness badge, will change to an [input-group](https://getbootstrap.com/docs/5.3/forms/input-group/)
- [ ] Allow-blank to implemented for all elements

**Reply by jonatanschroeder** [reply]
Adding #5540.

**Comment by eliotwrobson** [comment]
@firasm https://github.com/PrairieLearn/PrairieLearn/pull/6980 was just merged! We should be able to use this as a guide for the rest of the input elements.

**Reply by firasm** [reply]
I had to prove to myself it was real and it isn't deployed yet so I finally spent some time running the latest version locally and seeing the difference.

Such a small change, but I love it !

<img width="967" alt="Screenshot 2023-07-10 at 8 59 34 PM" src="https://github.com/PrairieLearn/PrairieLearn/assets/2507459/fdead4fa-a43d-4707-8a3d-02415084e34d">

Love it !
